app/models/draft_sequence.rb

```diff
@@ -9,14 +9,29 @@ def self.next!(user, key)
 
     return 0 if !User.human_user_id?(user_id)
 
-    h = { user_id: user_id, draft_key: key }
-    c = DraftSequence.find_by(h)
-    c ||= DraftSequence.new(h)
-    c.sequence ||= 0
-    c.sequence += 1
-    c.save!
-    DB.exec("DELETE FROM drafts WHERE user_id = :user_id AND draft_key = :draft_key AND sequence < :sequence", draft_key: key, user_id: user_id, sequence: c.sequence)
-    c.sequence
+    begin
+      ActiveRecord::Base.transaction(requires_new: true) do
```

Comment:
Would this not be a good fit for postgres' INSERT ON CONFLICT? You have manual sql below and I believe it would save you this transaction stuff.

Reply:
Yes, that's better. I can't quite do it all in one query because I want a RETURNING clause on the update, but definitely an improvement.

Reply:
Actually, maybe I can do it in one query.